core/variant/variant.h

```diff
@@ -290,6 +290,9 @@ class Variant {
 	bool is_one() const;
 	bool is_null() const;
 
+	Variant &operator[](const Variant &p_key) = delete;
+	const Variant &operator[](const Variant &p_key) const = delete;
```

Comment:
Add a comment, since its purpose is not obvious at a glance:
```suggestion
    // Make sure Variant is not implicitly cast when accessing it with bracket notation.
    // This prevents invalid code like this from compiling successfully (and failing at run-time):
    //
    //     Variant test;
    //     print_line(test["test"]);
	Variant &operator[](const Variant &p_key) = delete;
	const Variant &operator[](const Variant &p_key) const = delete;
```

Reply:
Sounds good, though I'd keep only the first line, we should avoid being too verbose in comments when it's not strictly needed to understand the API.

Could be:
```suggestion
	// Make sure Variant is not implicitly cast when accessing it with bracket notation (GH-49469).
	Variant &operator[](const Variant &p_key) = delete;
	const Variant &operator[](const Variant &p_key) const = delete;
```
So folks can check the PR for details.